mip/options.go

```diff
@@ -1,10 +1,203 @@
 package mip
 
-import "time"
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
 
-// Limits on the MIP solver, such as how long it can run for.
-type Limits struct {
+// SolveOptions are options that can be cionfigured for any solver.
+type SolveOptions struct {
 	// Duration is the maximum duration of the solver. A duration limit of 0 is
 	// treated as infinity.
-	Duration time.Duration `json:"duration" usage:"maximum duration of the solver. A duration limit of 0 is treated as infinity" default:"30s"`
+	Duration time.Duration `json:"duration" usage:"Maximum duration of the solver." default:"30s"`
+	// Verbosity of the solver in the console.
+	Verbosity Verbosity `json:"verbosity" usage:"{off, low, medium, high} Verbosity of the solver in the console." default:"off"`
+	// Mip-specific options.
+	Mip MipOptions `json:"mip" usage:"Options specific to MIP problems. Linear problems do not use these options."`
+	// Control options for the specific solver.
+	Control ControlOptions `json:"control" usage:"Options to control a specific solver, as defined by the provider."`
+}
+
+// MipOptions are options specific to MIP problems. LP problems do not use
+// these options.
+type MipOptions struct {
+	// Gap stopping criteria.
+	Gap GapOptions `json:"gap" usage:"Gap stopping criteria."`
+}
+
+// GapOptions specifies the gap stopping criteria.
+type GapOptions struct {
+	// Absolute gap.
+	Absolute float64 `json:"absolute" usage:"Absolute gap stopping value. If the problem is an integer problem the solver will stop if the gap between the relaxed problem and the best found integer problem is less than this value." default:"0.0"`
+	// Relative gap.
+	Relative float64 `json:"relative" usage:"Relative gap stopping value. If the problem is an integer problem the solver will stop if the relative gap between the relaxed problem and the best found integer problem is less than this value." default:"0.0"`
```

Comment:
Can we survive on the `0.0`? I think we ran into problems with this before due to numerical issues. E.g., Gurobi uses `1e-4` and `1e-10` at default.

Reply:
The templates had the gap set to 0, so I kept that. To your point, I think let's change it to something similar to your suggestion. Thanks.

Reply:
There was some reason for adapting them in the templates, I think.
But yeah, 0,0 seems pretty aggressive for defaults. :sweat_smile: 
Thank you! :hugs:

Reply:
I went with what HiGHS uses:
- 1e-6 for absolute
- 1e-4 for relative